java/src/jni/h5util.c

```diff
@@ -3607,6 +3869,74 @@ Java_hdf_hdf5lib_H5_H5export_1dataset(JNIEnv *env, jclass clss, jstring file_exp
         H5Fclose(file_id);
 } /* end Java_hdf_hdf5lib_H5_H5export_1dataset */
 
+/*
+ * Class:     hdf_hdf5lib_H5
+ * Method:    H5export_attribute
+ * Signature: (Ljava/lang/String;Ljava/lang/String;Ljava/lang/String;I)V
+ */
+JNIEXPORT void JNICALL
+Java_hdf_hdf5lib_H5_H5export_1attribute(JNIEnv *env, jclass clss, jstring file_export_name, jstring file_name,
+                                        jstring object_path, jint binary_order)
+{
+    const char *file_export = NULL;
+    const char *object_name = NULL;
+    const char *fileName    = NULL;
+    jboolean    isCopy;
+    herr_t      ret_val = FAIL;
+    hid_t       file_id = H5I_INVALID_HID;
+    hid_t       attr_id = H5I_INVALID_HID;
+    FILE *      stream  = NULL;
+
+    UNUSED(clss);
+
+    if (NULL == file_export_name)
+        H5_NULL_ARGUMENT_ERROR(ENVONLY, "H5export_attribute: file_export_name is NULL");
+
+    if (NULL == file_name)
+        H5_NULL_ARGUMENT_ERROR(ENVONLY, "H5export_attribute: file_name is NULL");
+
+    if (NULL == object_path)
+        H5_NULL_ARGUMENT_ERROR(ENVONLY, "H5export_attribute: object_path is NULL");
+
+    PIN_JAVA_STRING(ENVONLY, file_name, fileName, NULL, "H5export_attribute: file name not pinned");
+
+    if ((file_id = H5Fopen(fileName, (unsigned)H5F_ACC_RDWR, (hid_t)H5P_DEFAULT)) < 0)
+        H5_LIBRARY_ERROR(ENVONLY);
+
+    PIN_JAVA_STRING(ENVONLY, object_path, object_name, &isCopy, "H5export_attribute: object_path not pinned");
+
+    if ((attr_id = H5Aopen(file_id, object_name, H5P_DEFAULT)) < 0)
```

Comment:
I don't believe H5Aopen can open an attribute according to a given pathname. For example, if you ask for "/g1/attr1", this is going to try to open the attribute named "/g1/attr1" that is supposed to be attached to the root group of "file_id". You'll either need to split the string here manually, open the target object, and then use H5Aopen with just the attribute name, OR modify this routine to take an object path and an attribute name separately, and then just pass those to H5Aopen_by_name.

Reply:
I forgot about this function, it was a todo function.
I will remove it for now.